README.md
# kk_downloader

KK Downloader is a TYPO3 extensions which you can use as 
Download system with counter, simple category management, sorting criteria and page browsing in the LIST-view.

# New Extension Maintainer

Since 05.05.2019 we (jweiland.net) have taken over the extension key `kk_downloader` and updated this extension a lot.

Many thanks to Kurt Kunig for his great work.

# Upgrade

* We have moved the static extension TS-Template to new location: `Configuration/TypoScript`. That's why you have to 
  set this TS-Template again in your TS-Template record.
* We have changed Marker-based HTML-Template to Fluid. Please check your Template-Path and the Template itself.
  * In SingleView you can access download-record with `{download}`
  * In ListView you can access download-records with `{downloads}`
  * You have access to all FlexForm Settings with `{settings}`
  * We don't use extbase. So no lowerCamelCase available. Please use `ip_last_download` instead of `ipLastDownload`
* Go to InstallTool and start DB compare. We have changed column `clicks` to INT(10)
* We have changed localization handling of kk_downloader completely and make use of Core API now. So please check, if
  everything is still working for you.
* We have changed handling of TS-setting `downloadIcon`. If a path is set, it will be used for all download files.
  If it is not set (default) we will use TYPO3 internal icons to show file ext related icons. It's not possible
  to set downloadIcon to a Path anymore.
* Please check `Documentation/ChangeLog/Index.rst` to see all changes.


ChangeLog
2017-05-16  Kurt Kunig <kurt.kunig@kupix.de>
  * Add error, if single view was called without UID in URI
  * Add option to FlexForm to show/hide pagebrowser
  * Update labels in FlexForm (showFileMDate/selectCategory)
  * Move CSS from TS-Template into own CSS file
  * Little Template and CSS modifications
  * Implement crDate to template back again

2019-05-09  Stefan Froemken <projects@jweiland.net>
  * Extension key kk_downloader was transferred to jweiland.net
  * Update version number to 4.0.0
  * Change Template to Fluid
  * Move marker based logic to Fluid-Template
  * Change default template to EXT:kk_downloader/Resources/Private/Templates/MainTemplate.html
  * Collect all FlexForm configuration in $this->settings
  * Assign downloads variable to ListView
  * Assign download variable to SingleView
  * $this->settings is available in Fluid, too
  * Removed all PHP code for translation handling and rebuild it with TYPO3 API
  * Moved all TCA from tca.php into Configuration/TCA
  * Moved all TCA changing PHP into Configuration/TCA/Overrides/*
  * Moved pi-class to Classes/Plugin
  * Moved additional fields UserFunc class to Classes/UserFunc
  * Changed yes/no selectboxes in FlexForm to Checkboxes
  * Change PHP code to be PSR-2 compatible
  * Add strict_types where possible
  * Add composer.json. So it's now available over packagist, too.
  * Set TYPO3 compatibility to 8.7 and 9.5. Yes, we have removed compatibility to TYPO3 7.6
  * Changed column "clicks" from tinytext to int(10)
  * Moved all DB Queries to non-extbase-based Repositories
  * Changed all DB-Queries to Doctrine
  * Add Namespacing and correct PHP DocHeaders
  * Simplify many if-conditions (don't test on empty() and empty string)
  * Removed all Debugging-Output. Please use xdebug or f:debug or Admin-Panel or...
  * Change pi_getLL to LocalizationUtility to use new path of language files
  * Change language files to XLF
  * Divide Languagefile to locallang, locallang_db and FlexForm
  * Change IMAGE() call to cObjGetSingle()
  * Remove or change many deprecated code
  * Create date with f:format.date in template directly
  * Moved all Icons to Resources/Public/Icons
  * Moved TS-Template from styles/css to Configuration/TypoScript
  * Remove imageDistance, as the result "$imagewidth" was not used
  * Change downloadIcon handling. If not set, use Icons of Core, else use Icon of DownloadIcon

2017-06-02  Kurt Kunig <kurt.kunig@kupix.de>
  * in class.tx_kkdownloader_addFieldsToFlexForm.php ergänzt: Kategorien alphabetisch sortiert!
   Danke für den Hinweis an Michael Lausberg, FH Aachen
  * Liste der Kategorien ebenfalls sortiert nach Kategorienamen anstatt Erstelldatum

2016-01-25  Kurt Kunig <kurt.kunig@kupix.de>
   * Ready for TYPO3 Version 6 up to 7.9.99

2015-07-14 Kurt Kunig <kurt.kunig@kupix.de>
  * make the ext. running under TYPO3 until V6.2.99
  * modified template-path handling in the flexform

2012-07-12 Kurt Kunig <kurt.kunig@kupix.de>
  * make the ext. running under TYPO3 V4.7 (replaced deprecated old functions: ->sql(... to ->sql_query(...)
  * possibility to crop the download-file description

2011-12-22 Kurt Kunig <kurt.kunig@kupix.de>
  * make the ext. running under TYPO3 V4.6 (replaced deprecated old functions)

2011-09-13 Kurt Kunig <kurt.kunig@kupix.de>
  * correction of the mimetype function (wrong allocation)

2011-05-24 Kurt Kunig <kurt.kunig@kupix.de>
  * there was no correct where function in categories if a cat-set is set (e.g. "2,1")
  * Thanks to Michael Hilgers <mhil@mhil.at> for the new statement: [$where2 = " AND (find_in_set('$cat',tx_kkdownloader_images.cat )) >0";] in line 253

2011-05-19 Kurt Kunig <kurt.kunig@kupix.de>
  * there was no correct select in categories if more than 9 cats exist
  * Thanks to Michael Hilgers <mhil@mhil.at> for this correction: [$where2 = " AND ($cat IN (tx_kkdownloader_images.cat))";] in line 251

2011-05-03 Kurt Kunig <kurt.kunig@kupix.de>
  * marker ###FILESIZE### was shown instead of nothing if filesize was not desired in the flexform

2010-12-20 Kurt Kunig <kurt.kunig@kupix.de>
  * download-objects can now be sorted in the item-storage-area with the well-known "up"- and "down"-buttons
  * additional sort-select in the plugin for items order in the FE: "by sorting in sysfolder"

2010-02-24 Kurt Kunig <kurt.kunig@kupix.de>
  * download-objects can now be stored in a normal page, not only in a sysfolder (that's simpler for e.g. 1 DL-file)

2010-01-15 Kurt Kunig <kurt.kunig@kupix.de>
  * some minor cosmetics (e.g. selection fileModificationDate = no, date or date and time in the flexform...)

2009-11-25 Kurt Kunig <kurt.kunig@kupix.de>
  * BUG FIX of a security bug

2009-10-31 Kurt Kunig <kurt.kunig@kupix.de>
  * page-browsing in LIST-view is now integrated if wanted. New markers in the HTML-template are ###RESULT_COUNT### (like Object 1 to 10 of 24), ###LINK_PREV### (e.g. previous), ###PAGES### (e.g. page 1, page 2, page 3) and ###LINK_NEXT### (e.g. next)
  * All markers are inserted in the default template “kk_download_template.html” of the extension

2009-09-09 Kurt Kunig <kurt.kunig@kupix.de>
  * flexible rendering of file-item data, marker ###LINKS### can be either be used as preformatted HTML-code or (new) each item quite alone. This depends on an additional template part in the HTML-template
  * Nearly all in the code preformatted items are now “clean”
  * Text-markers are included: general name = ###TEXT_KEYWORD### corresponding to the locallang-parameter txtKEYWORD.
  * Missing a correct HTML-Template puts out an error-message in the content-field.
  * Now the ext. is completely multilingual (I hope)

2009-03-17 Kurt Kunig <kurt.kunig@kupix.de>
  * new extension parameter “parseFunc” added

2009-03-15 Kurt Kunig <kurt.kunig@kupix.de>
  * flexform-field “Show IP-address of last download” added

ext_emconf.php
<?php

/***************************************************************
 * Extension Manager/Repository config file for ext "kk_downloader".
 *
 * Auto generated 23-09-2016 17:16
 *
 * Manual updates:
 * Only the data in the array - everything else is removed by next
 * writing. "version" and "dependencies" must not be touched!
 ***************************************************************/

$EM_CONF[$_EXTKEY] = [
    'title' => 'Simple download-system with counter and categories',
    'description' => 'Download system with counter, simple category management, sorting criteria and page browsing in the LIST-view. Configuration via flexforms and HTML template. (example: http://kupix.de/downloadlist.html)',
    'category' => 'plugin',
    'version' => '5.0.0',
    'state' => 'stable',
    'uploadfolder' => true,
    'createDirs' => '',
    'clearcacheonload' => false,
    'author' => 'Stefan Froemken',
    'author_email' => 'projects@jweiland.net',
    'author_company' => 'jweiland.net',
    'constraints' => [
        'depends' => [
            'typo3' => '8.7.14-9.5.99',
        ],
        'conflicts' => [
        ],
        'suggests' => [
        ],
    ],
];
